applications/RANSApplication/custom_processes/auxiliary_processes/rans_check_vector_bounds_process.cpp

```diff
@@ -113,36 +115,49 @@ void RansCheckVectorBoundsProcess::Execute()
         break;
     }
 
-    double min_value = 0.0;
-    double max_value = 0.0;
-    bool is_initialized = false;
+    double min_value = std::numeric_limits<double>::max();
+    double max_value = std::numeric_limits<double>::min();
 
     const int number_of_nodes = r_nodes.size();
-    for (int i_node = 0; i_node < number_of_nodes; ++i_node)
-    {
-        const ModelPart::NodeType& r_node = *(r_nodes.begin() + i_node);
-        const array_1d<double, 3>& vector_value =
-            r_node.FastGetSolutionStepValue(vector_variable);
+    const int number_of_threads = OpenMPUtils::GetNumThreads();
+    OpenMPUtils::PartitionVector node_partition;
+    OpenMPUtils::DivideInPartitions(number_of_nodes, number_of_threads, node_partition);
+
+    Vector max_values(number_of_threads, max_value);
+    Vector min_values(number_of_threads, min_value);
 
-        double current_value = 0.0;
-        for (int dim = 0; dim < 3; ++dim)
-            current_value += std::pow(vector_value[dim] * vector_weights[dim], 2);
+#pragma omp parallel
```

Comment:
probably better to use a [reduction](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/solving_strategies/convergencecriterias/displacement_criteria.h#L354) here ;)

Reply:
Reduction is quite troublesome with VS

Reply:
thats why I was suggesting sth that is in the core and hence works in VS
check the link ;)

Reply:
Is there a way to use min, max reductions in windows with VS? I had a compilation error before, so I had to change it to this.